actions/substitute-envs/action.yaml

```diff
@@ -0,0 +1,35 @@
+name: "Substitute envs"
+description: "Recursively substitute environment variables in a given directory"
+inputs:
+  path:
+    description: "Path to the directory to substitute environment variables in"
+    required: true
+outputs:
+  path:
+    description: "Path to the temporary directory after substitution"
+    value: ${{ steps.substitute.outputs.path }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Substitute envs
+      id: substitute
+      shell: bash
+      run: |
+        function substitute() {
+          # create (sub-)directories
+          mkdir -p "${{ runner.temp }}/$(dirname "$1")"
```

Comment:
consider using `mktemp -d`

Reply:
not sure there is an actual advantage in this case. it's usually safer to use `mktemp` than `mkdir` in scripts. feel free to dismiss if you think there aren't any concerns with using `mkdir`